examples/graphene_apps/python_worker/avalon_worker/work_order_processor.py

```diff
@@ -0,0 +1,364 @@
+
+    def process_work_order(self, msg):
+        """
+        Process Avalon work order or signup request.
+
+        Parameters :
+            msg: JSON formatted string containing the request to execute.
+        Returns :
+            JSON RPC response containing result or error.
+        """
+        try:
+            json_msg = json.loads(msg)
+        except json.decoder.JSONDecodeError:
+            err_code = WorkerError.INVALID_PARAMETER_FORMAT_OR_VALUE
+            err_msg = "Invalid JSON"
+            error_json = jrpc_utility.create_error_response(err_code,
+                                                            0,
+                                                            err_msg)
+            return json.dumps(error_json)
+        method_name = json_msg["method"]
+        params = json_msg["params"]
+        if (method_name == "ProcessWorkerSignup"):
+            output = self._process_worker_signup()
+        elif (method_name == "ProcessWorkOrder"):
+            output = self._process_work_order(params)
+        else:
+            err_code = WorkerError.INVALID_PARAMETER_FORMAT_OR_VALUE
+            err_msg = "Unsupported method"
+            error_json = jrpc_utility.create_error_response(err_code,
+                                                            0,
+                                                            err_msg)
+            output = json.dumps(error_json)
+
+        return output
+
+# -------------------------------------------------------------------------
+
+    def _process_worker_signup(self):
+        """
+        Process signup request and returns worker details.
+
+        Returns :
+            JSON signup response conatining worker details.
+        """
+        signup_data_json = {}
+        signup_data_json["sealed_data"] = ""
+        signup_data_json["verifying_key"] = \
+            self.worker_public_sign_key.decode("ascii")
+        signup_data_json["encryption_key"] = \
+            self.worker_public_enc_key.decode("ascii")
+        signup_data_json["encryption_key_signature"] = \
+            self.worker_public_enc_key_sign.hex()
+        signup_data_json["mrenclave"] = \
+            self._get_mrenclave()
+        signup_data_json["quote"] = \
+            self._get_quote()
+
+        return json.dumps(signup_data_json)
+
+# -------------------------------------------------------------------------
+
+    def _process_work_order(self, input_json_str):
+        """
+        Process Avalon work order and returns JSON RPC response
+
+        Parameters :
+            input_json_str: JSON formatted work order request string.
+                            work order JSON if formattted as per
+                            TC spec ver 1.1 section 6.
+        Returns :
+            JSON RPC response containing result or error.
+        """
+        try:
+            input_json = json.loads(input_json_str)
+        except json.decoder.JSONDecodeError:
```

Comment:
Double check for None else runtime issues could be seen.

Reply:
Yes. Better to catch other exception with Exception